tools/awps-tunnel/client/src/panels/RequestHistory.tsx

```diff
@@ -126,28 +128,65 @@ export function RequestHistory(props: RequestHistoryProps) {
   );
 }
 
+const renderContent = (rawString: string, index: number): ReactNode => {
+  try {
+    const parsedJson = JSON.parse(rawString.substring(index));
+    return <div><ReactJson src={parsedJson} collapsed={1}/></div>;
+  } catch (error) {
+    return <div className="m-2" style={{ whiteSpace: "pre-wrap" }}>
+      {rawString.substring(index)}
+    </div>;
+  }
+};
+
 function Details({ item }: { item?: HttpHistoryItem }) {
   if (!item) return <></>;
-  let requestTabItems = [
+  const requestStartIndex: number = item.requestRaw.indexOf('\n\n');
+  const requestTabItems = [
     {
-      title: "Request Details",
-      content: (
-        <div className="m-2" style={{ whiteSpace: "pre-wrap" }}>
-          {item.requestRaw}
+      title: "Formatted Request Details",
+      content:(
+        <div>
+          <label style={{fontWeight:"bold"}}>Header</label>
+          <div className="m-2" style={{ whiteSpace: "pre-wrap" }}>
+            {item.requestRaw.substring(0,requestStartIndex)}
+          </div>
+          <label style={{fontWeight:"bold"}}>Content</label>
+          {renderContent(item.requestRaw, requestStartIndex)}
         </div>
-      ),
+      )
     },
-  ];
-  let responseTabItems = [
     {
-      title: "Response Details",
+      title: "Raw Request Details",
+      content: <div className="m-2" style={{ whiteSpace: "pre-wrap" }}>
+        {item.requestRaw}
+      </div>
+    }
+  ]
+  const responseTabItems = [
+    {
+      title: "Formatted Response Details",
       content: (
+      <div>
+        <label style={{fontWeight:"bold"}}>Header</label>
+        <div className="m-2" style={{ whiteSpace: "pre-wrap" }}>
+          {item.responseRaw?.substring(0,item.responseRaw?.indexOf('\n\n'))}
```

Comment:
`\r\n` is also possible depending on different platforms

we can wrap the processing into a utils method to return the structured header and body instead, maybe next to utils's `dumpRawRequest`: https://github.com/Azure/azure-webpubsub/blob/main/tools/awps-tunnel/server/util.ts

Reply:
`dumpRawRequest` is in server repo, is that ok to call some method that's located such far away from itself?

Reply:
resolved with splitter function in new commit.